### ISSUE TITLE
Crash on Linux

### ISSUE THREAD

**Issue by JanThomas118** [supsm/MCPPPP#17]
**Describe the bug**
The Program crashes on Linux when starting the conversion

**Important stuff**
 - OS: Linux, Fedora 35
 - newest

**Resourcepack**
Misas realistic https://www.curseforge.com/minecraft/texture-packs/misas-realistic

**Logs**
[jan@janpc Downloads]$ ./MCPPPP-linux-gui 
/builddir/build/BUILD/gcc-11.2.1-20211203/obj-x86_64-redhat-linux/x86_64-redhat-linux/libstdc++-v3/include/bits/basic_string.h:1114: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::front() [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference = char&]: Assertion '!empty()' failed.

**Additional context**

**Comment by supsm** [comment]
By logs, I meant mcpppp logs :p
Can you send those as well? It should be mcpppp-log.txt

**Comment by JanThomas118** [comment]
`[13:00:20] No settings found
[13:00:20] MCPPPP 0.5.7 (GUI)
[13:00:20] Os: Linux

[13:00:20] autoDeleteTemp  false
[13:00:20] pauseOnExit     true
[13:00:20] log             mcpppp-log.txt
[13:00:20] timestamp       true
[13:00:20] outputLevel     3
[13:00:20] logLevel        1
[13:00:20] autoReconvert   false
[13:00:20] fsbTransparent  true


[13:00:22] Path: /home/jan/.local/share/multimc/instances/1.18/.minecraft/resourcepacks
[13:00:26] Conversion Started
[13:00:32] Extracting Misa_1.18a.zip
[13:00:38] FSB: Converting Pack Misa_1.18a
[13:00:38] FSB: Converting sky1.properties`

I was trying to convert the skybox textures, i pressed start, it then did something for 10 seconds and then crashed

**Comment by supsm** [comment]
Thanks, now can you send the resource pack you were trying to convert

**Comment by JanThomas118** [comment]
https://www.curseforge.com/minecraft/texture-packs/misas-realistic/files/3543877

**Comment by supsm** [comment]
I can't reproduce this issue. It took forever to zip and unzip, but it didn't crash. Can you maybe try unzipping the pack first, then converting the unzipped one?

Also if you need the pack urgently, I've converted it for you: https://www.mediafire.com/file/5ik2r5fk3txo7uv/Misa_1.18a.zip/file

**Comment by DownloadableFox** [comment]
Here's the backtrack, found the error.
Creating a pull request fixing the issue right now.
```(gdb) backtrace
#0  __memmove_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:502
#1  0x00005555555c1bb4 in std::char_traits<char8_t>::move (__s1=0x7fffe02234b0 "/home/fox/Deskto", __s2=0x7fffe022350c "mcpppp-temp", __n=18446744073709551615) at /usr/include/c++/11/bits/char_traits.h:645
#2  0x00005555555e56b8 in std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::_S_move (__d=0x7fffe02234b0 "/home/fox/Deskto", __s=0x7fffe022350c "mcpppp-temp", __n=18446744073709551615)
    at /usr/include/c++/11/bits/basic_string.h:368
#3  0x00005555555e2262 in std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::_M_erase (this=0x7ffff6405e30, __pos=0, __n=92) at /usr/include/c++/11/bits/basic_string.tcc:330
#4  0x00005555555df75f in std::__cxx11::basic_string<char8_t, std::char_traits<char8_t>, std::allocator<char8_t> >::erase (this=0x7ffff6405e30, __first=47 '/', __last=109 'm') at /usr/include/c++/11/bits/basic_string.h:1872
#5  0x0000555555603d75 in vmt::png (path=filesystem::path "mcpppp-temp/Misa_1.18a"<error reading variable: Cannot access memory at address 0x7ffff2cf29c000>, optifine=true, newlocation=false, zip=true, entry=...) at ../src/vmt.cpp:138
#6  0x000055555560cdc2 in vmt::convert (path=filesystem::path "mcpppp-temp/Misa_1.18a"<error reading variable: Cannot access memory at address 0x7ffff2cf29c000>, filename="Misa_1.18a.zip", info=...) at ../src/vmt.cpp:835
#7  0x0000555555689f2a in mcpppp::convert (path=filesystem::path "/home/fox/Downloads/test/Misa_1.18a.zip"<error reading variable: Cannot access memory at address 0x5555558c987000>, dofsb=true, dovmt=true, docim=true)
    at ../src/utility.cpp:786
#8  0x00005555556ecbd2 in mcpppp::guirun () at ../src/gui.cpp:77
#9  0x00005555556de989 in std::__invoke_impl<void, void (*)()> (__f=@0x5555559ba148: 0x5555556eca3d <mcpppp::guirun()>) at /usr/include/c++/11/bits/invoke.h:61
#10 0x00005555556de941 in std::__invoke<void (*)()> (__fn=@0x5555559ba148: 0x5555556eca3d <mcpppp::guirun()>) at /usr/include/c++/11/bits/invoke.h:96
#11 0x00005555556de8ee in std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ul> (this=0x5555559ba148) at /usr/include/c++/11/bits/std_thread.h:253
#12 0x00005555556de8a8 in std::thread::_Invoker<std::tuple<void (*)()> >::operator() (this=0x5555559ba148) at /usr/include/c++/11/bits/std_thread.h:260
#13 0x00005555556de824 in std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run (this=0x5555559ba140) at /usr/include/c++/11/bits/std_thread.h:211
#14 0x00007ffff7c88694 in ?? () from /lib/x86_64-linux-gnu/libstdc++.so.6
#15 0x00007ffff791c947 in start_thread (arg=<optimized out>) at pthread_create.c:435
#16 0x00007ffff79aca44 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:100```

**Comment by supsm** [comment]
Should be fixed by #23
